pkg/controller/plan/adapter/ocp/builder.go

```diff
@@ -230,6 +230,12 @@ func (r *Builder) Tasks(vmRef ref.Ref) (list []*planapi.Task, err error) {
 	return
 }
 
+func (r *Builder) InstanceTypeName(vmRef ref.Ref) (name string, err error) {
+	// The VM is build from configuration, we don't need the instancetype
+	err = liberr.New("instancetypes are not used by this provider")
```

Comment:
did you check what happens in this case of migration of an VM with instance type from one cluster to another, in terms of the instance type specification that is exposed by CDI? it can preserve the original instance type that the vm was based on (which is fine) or "collapse" the properties into the exposed configuration (which is not that great since we lose the instance type that the vm originally had, but on the other hand - if the instance type preferences are different in both clusters than it might be better). need to decide on what's the right direction here

Reply:
Instancetypes and preferences can vary between clusters. Also, a VM always uses a specific ControllerRevision of an instancetype or preference. If you move the VM definition over to another cluster without moving the ControllerRevision of instancetype/preference too, the VM cannot start. 

Options would be:

- Clearing the `revisionName` of the instancetype or preference (Caution: Instancetype or preference on dest cluster might look different or are nonexistent at all)
- Use the Expand endpoint of the KubeVirt API to expand the use instancetype/preference into the VM definition (you lose the abstraction, your VM definition becomes larger)

Reply:
> Options would be:
> 
> * Clearing the `revisionName` of the instancetype or preference (Caution: Instancetype or preference on dest cluster might look different or are nonexistent at all)
> * Use the Expand endpoint of the KubeVirt API to expand the use instancetype/preference into the VM definition (you lose the abstraction, your VM definition becomes larger)

I don't mind expanding (yes, that's the correct term, I wrote "collapse" before but meant "expand") the configuration since the VM is already provisioned (so the abstraction doesn't give us much and I don't think the length of the VM definition would be important at this point).

but there's an argument saying that typical VMs in kubevirt are based on instancetype/preference and we don't want VMs that are imported from MTV to be different...

> * create new preference by forklift, copying the exact source preference.

the problem with this approach is that the instancetype/preference might already exist on the desintation - so we may end up with centos-from-cluster-x, centos-from-cluster-y, etc - which is not that great

@fabiand we talked about this in the context of importing appliance from OVA - the question is how important is it for you to avoid having VMs that are not based on instancetypes/preferences, because AFAIU it doesn't give us much value for VMs that are already provisioned in kubevirt